### PR TITLE
Added initial mypy type checking

### PR DESCRIPTION
**Pull request by sebastian-philipp**
This is very minimalistic without any type annotations,
but in general this is *already as it is* super helpful for uses.

Signed-off-by: Sebastian Wagner <sebastian@inter.link>

### CODE REVIEW

File: safedelete/models.py [hunk 1]
```diff
@@ -1,7 +1,6 @@
-from collections import Counter
-from collections import defaultdict
-from itertools import chain
 import warnings
+from collections import Counter, defaultdict
+from itertools import chain
```
Comment (sebastian-philipp) [inline safedelete/models.py:3]: can't explain why this change happened. Any ideas?

Comment (Gagaro) [reply safedelete/models.py:3]: Do you have isort automatically running on save?

**Comment by Gagaro** [comment]
Thanks, that's great :+1: . I'll try to add some typings.

File: safedelete/models.py (hunk 1 above)
Comment (Gagaro) [reply safedelete/models.py:3]: isort is actually always called with tox as it is configured. That's probably why.

**Comment by Gagaro** [comment]
I added some typings in https://github.com/makinacorpus/django-safedelete/commit/05cfa2c8f1bb50f0606b79a9a760c7ca927d55d4 if you want to take a look at them and tell me what you think about them.

**Comment by Gagaro** [comment]
There are some issue with Python 3.6 I'll check later on https://github.com/makinacorpus/django-safedelete/runs/7861846714?check_suite_focus=true

**Comment by sebastian-philipp** [comment]
added a few comments. But https://github.com/makinacorpus/django-safedelete/commit/05cfa2c8f1bb50f0606b79a9a760c7ca927d55d4 looks great!
